### DISCUSSION TITLE
Morph test?

### DISCUSSION THREAD

**Discussion by newbrain** [pyocd/pyOCD#1052] [Q&A] [answered]
@flit kindly advised me to ask questions here, so I'll bite:

I see that my very simple PR #1048, is blocked on the /morph test.

That would seem to be a fairly common issue, by looking at previous PRs.
Is there a description of what that means, and are there any actions I should take either before submitting the PR to avoid it or afterwards to correct the issue?

**Comment by flit** [comment, marked as answer]
Hi @newbrain 😄 

`/morph test` is a trigger comment for the [Jenkins GitHub Pull Request Builder plugin](https://plugins.jenkins.io/ghprb/) for our CI system. Since the pyocd CI tests involve running community-submitted code directly on the test machines, there is the possibility for someone to do something malicious. Thus we have a manual trigger to kick off the test after reviewing the code changes.

The trigger comment `/morph test` comes from an old version of Mbed OS with an internal codename of Morphius. (pyocd used to be developed by the Mbed team, though it is now independent.) Otherwise the Jenkins plugin allows any comment you want to use.

The current CI system is old and having lots of trouble, which is why there are so many failures. Test boards have been highly unstable lately. Not being able to go up to the office due to continued pandemic restrictions makes it pretty difficult to diagnose and fix the problems. So in most cases I just manually review the test results to look for new failures, and then use admin privileges to force merge the PR.

I'm not too concerned with the current CI system's troubles because we are in the process of building an entirely new CI system at Arm's offices in Cambridge, UK to replace the current one. The goal is to have it up and running within 2-3 months. We plan to replace Jenkins with a cloud based CI pipeline server such as GitHub Actions or Azure Pipelines, which has the major benefit of allowing contributors to see the full logs from PR tests.

(Both the current and new CI systems are responsible for testing both pyOCD and DAPLink.)

I hope that helps answer your questions!

**Comment by newbrain** [comment]
Thanks, clear and instrucitve answer!
Then it would seem I did nothing wrong, oh well, there will be more opportunities. 😉
